dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java

```diff
@@ -641,35 +642,75 @@ protected void refreshTargetEntry(String path) {
 
     @Override
     protected void handlePublishUnpublish(String entryPath, String newName, boolean unpublishRequest) {
-        Workflow existingWorkflow;
+        Workflow existingWorkflow = null;
         boolean isPublished = false;
+
+        // Cannot be making an unpublish request with a specified new name
+        assert (!(unpublishRequest && newName != null));
+
         try {
             existingWorkflow = workflowsApi.getWorkflowByPath(entryPath, null, false);
             isPublished = existingWorkflow.isIsPublished();
         } catch (ApiException ex) {
-            exceptionMessage(ex, "Unable to publish/unpublish " + newName, Client.API_ERROR);
+            exceptionMessage(ex, "Unable to " + (unpublishRequest ? "unpublish " : "publish ") + entryPath, Client.API_ERROR);
+        }
+
+        if (existingWorkflow == null) {
```

Comment:
Same as other comment; I don't think you need this block.

Reply:
Agreed. Same as above.